### PR TITLE
Removed try-catch from controllers and replaced List to IList and some code refactoring

### PR DESCRIPTION
**Pull request by BohdanBBG**
Removed unnecessary try-catch blocks from controllers and replaced List to IList in return type in some methods, and separated logical blocks in unit test for controllers.

### CODE REVIEW

**Review by sober607 — CHANGES_REQUESTED** [review]
There is many changes in all files related to many tasks. Please separate.
Also in appsettings.json LIFETIME should be 720 (accrd to expert requirement).

File: CharlieBackend.Api.UnitTes/Controllers/LessonsControllerUnitTest.cs [hunk 1]
```diff
@@ -14,17 +14,25 @@ public class LessonsControllerUnitTest
...
             var toCompare = objectResult.Value as List<LessonModel>;
             var actualResult = await getLessons();
+
+            //Assert
+
             Assert.Equal(toCompare.Count, actualResult.Count);
         }
 
-        public async Task<List<LessonModel>> getLessons() 
+        public async Task<IList<LessonModel>> getLessons()
```
Comment (whereismyroot) [inline CharlieBackend.Api.UnitTes/Controllers/LessonsControllerUnitTest.cs:35]: Use PascalCase in all methods

**Review by whereismyroot — APPROVED** [review]

File: CharlieBackend.Api.UnitTes/Controllers/LessonsControllerUnitTest.cs (hunk 1 above)
Comment (BohdanBBG) [reply CharlieBackend.Api.UnitTes/Controllers/LessonsControllerUnitTest.cs]: Done

**Review by SnipeFrost — APPROVED** [review]

**Review by sober607 — APPROVED** [review]
Ok.

**Pull request merged**